### PR TITLE
Ye updates

### PR DESCRIPTION
**Pull request by YeeeHeeee**
updates 6 files,

### CODE REVIEW

**Comment by gputtley** [comment]
Please can you remove the config file. These in general should not be pushed. Can you all remove the commented out lines of code. It is also unclear to me why you have made these changes in histogram_metrics, can you explain these?

**Comment by YeeeHeeee** [comment]
changes in histogram_metrics are to be compatible with the classifer metrics calculation

**Comment by gputtley** [comment]
Please can you be more specific, why is this change needed for classifier metrics?

**Comment by YeeeHeeee** [comment]
To use a fixed number of bins so that histograms remain consistent in binning.

**Comment by gputtley** [comment]
Again, please can you be more elaborate on why this is needed. 

It seems to me the change is (a) not necessary and (b) not valid.

(a) For the use of 'wt_total', I would either change the weight before the histogram metrics so it is called 'wt' or add an option in the histogram metrics to parse the weight name. This 'type' option is not general enough.

(b) For the use of n_bins over bins, what you have implemented is incorrect. You can **never** compare histograms with different bin edges. When you get a histogram with an integer for the bins entry, it will linearly space the bins between the minimum and maximum values. The minimum and maximum between the sim and synth datasets will be different, so you will get different bin edges. This is why in the code, it gets the bins edges from the synth dataset and applies those bins edges when getting the sim hist. It is important that you understand this point, I remember you have made this mistake in the past as well, so if you do not, then please ask.

**Comment by YeeeHeeee** [comment]
For (a) using `wt_tot` is simply to prevent confusion with the default `wt`, as the shifted dataset also contains a `wt` variable.
For (b) nominal dataset is normalised, so for example `CombinedSubJets_mass` values roughly between -3 to 3; the shifted dataset uses physical units, so values roughly between 80 to 500. Then the numerical ranges are completely different, and the bin edges derived from the nominal dataset do not cover the range of the shifted dataset.

**Comment by gputtley** [comment]
ok for (a) I suggest adding options to the histogram metrics called sim_wt_name and synth_wt_name, default to "wt" so you do not break any other code. Then you can parse synth_wt_name="wt_total" when you run the code.

For (b) I do not understand why one class is standardised and the other isn't, they should both come from the same dataset (train or test) just with classifier_truth = 0 and 1 and both should be standardised?

**Comment by gputtley** [comment]
I also just noticed in classifier_performance_metrics you build the network twice, this should not be needed.

**Comment by YeeeHeeee** [comment]
I think we discussed this before: since the network is built inside the Run function, it needs to be built again for DoHistogramMetrics. function We could save the model from Run locally and reuse it in DoHistogramMetrics ? But I don’t think building the model twice is why one is standardised and the other isn’t.

**Comment by gputtley** [comment]
I think I misunderstood before, instead of all this:

https://github.com/gputtley/INNFER/blob/538057cafe38e3dcfc9136e4a4a88dcc553909d6/python/runner/classifier_performance_metrics.py#L192-L203

you can just use self.network

**Comment by YeeeHeeee** [comment]
Building the model twice is not the reason for one being standardised and the other not.

**Comment by gputtley** [comment]
This looks much better. Here are my final comments before merging:

1) In classifier_performance_metrics.py you will need to make it work for both train and test datasets (or rather self.histogram_datasets and self.loss_datasets), at the moment it looks very hard coded for only the train dataset

2) Remove commented out lines of code everywhere in the pull request. Also remove all commented lines from the config input to the module in innfer.py

3) I think the defaults for the argparse (to innfer.py) should just be 'loss,histogram', I would not make the separation between chi_squared and kl_divergence here, as they are both histogram metrics. Also, multidim is not yet implemented.

**Comment by YeeeHeeee** [comment]
1. But didn’t we agree that we only need to test the train file, and there’s no need to use self.loss_datasets?
2. If we don’t explicitly define chi_squared and kl_divergence, those steps don’t seem to run. Also, we agreed to ignore multidim for now—we just need to set up chi_squared and kl_divergence

**Comment by gputtley** [comment]
1) We agreed we only need the train and test datasets and not the validation datasets. You should calculate histogram metrics for what is in self.histogram_datasets. I see now that you did not implement the loss metric. This can be done in a future pull request, but for now clear up anything that is not used, so not to cause confusion.

2) When I say the default args should just be loss,histogram, I do not mean just change the code exactly how I said. It means make the code work in a way that those options are the only two options. As I said before it makes no sense having histogram, chi_squared and kl_divergence as the input, as the latter two are both histogram metrics! I think it is fine for both chi_squared and kl_divergence to be ran when you parse histogram as the option. Finally, as I said in point 1, please remove any options/settings to the code for stuff which is not yet implemented.

I would really like to move on to hyperparameter tuning next week, and the remaining changes should not be difficult to implement. Can you work to get these done so I can review and merge by tomorrow?

**Comment by YeeeHeeee** [comment]
I have updated the file.

**Comment by gputtley** [comment]
Thanks! One more really annoying comment. Please can you change the indent in classifier_performance_metrics to be 2 instead of 4, to be the same as the rest of the code in the repository.

**Comment by gputtley** [comment]
Thanks! Good job with the code. I will merge now.
